### DISCUSSION TITLE
Could not find any data to train the Entity Linker on.

### DISCUSSION THREAD

**Discussion by shrinidhin** [explosion/spaCy#10704] [Help: Coding & Implementations] [answered]
Hi!I have been trying to build an Entity Linker model. I have been following the [demo project](https://github.com/explosion/projects/tree/v3/tutorials/nel_emerson) along with the [video](https://www.youtube.com/watch?v=8u57WSXVpmw). I have annotated my data and created my training and dev set in spacy format as well. When I try to train the EntityLinker, I get an error saying that there is no training data. 

```
C:\Users\Shrinidhi\Desktop\Sentiment Analysis\sentiment\lib\site-packages\spacy\pipeline\entity_linker.py:276: UserWarning: [W093] Could not find any data to train the Entity Linker on. Is your input data correctly formatted?
  warnings.warn(Warnings.W093.format(name="Entity Linker"))
```

Which is absurd because I have followed the format of scripts as given for spacy's v3 config system. The only difference is I have multiple entities for each `doc` object each assigned different kb_ids which I have set accordingly. What could be the issue?

**Comment by polm** [comment, marked as answer]
It looks like that warning can only be issued in v1 of the EntityLinker, and it only comes up if you have a batch with no annotations. This could happen if some of the docs in your training set happen to not have any entities in them, or the entites present have no entries in your knowledge base. If it is happening only sometimes it is not necessarily a problem, and could be caused by an unlucky batch. (Do note that unlike some other components, the EntityLinker isn't really helped by data that correctly has no annotations.)

If you can share an example document / knowledge base that by themselves have this problem we could look at it more. Alternately, you could upgrade to v2 of the EntityLinker architecture - you should just need to change the version in your config if you're using a recent enough version of spaCy.
